src/transport/raw/TCP.cpp

```diff
@@ -40,10 +40,10 @@ namespace {
 using namespace chip::Encoding;
 
 // Packets start with a 16-bit size
-constexpr size_t kPacketSizeBytes = 2;
+constexpr size_t kPacketSizeBytes = sizeof(uint32_t);
 
 // TODO: Actual limit may be lower (spec issue #2119)
-constexpr uint16_t kMaxMessageSize = static_cast<uint16_t>(System::PacketBuffer::kMaxSizeWithoutReserve - kPacketSizeBytes);
+constexpr uint32_t kMaxTCPMessageSize = static_cast<uint32_t>(CHIP_CONFIG_MAX_LARGE_PAYLOAD_SIZE_BYTES - kPacketSizeBytes);
```

Comment:
I don't follow this.  we're saying the max message size is smaller than the max payload size?  Note that the old code _did_ make sense, in that it had the message size being smaller than the max size of a buffer we might be stuffing the message + length into.

Reply:
Renamed CHIP_CONFIG_MAX_LARGE_PAYLOAD_SIZE_BYTES to CHIP_SYSTEM_CONFIG_MAX_LARGE_BUFFER_SIZE_BYTES